nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java

```diff
@@ -158,34 +173,37 @@ protected void init(final ProcessorInitializationContext context) {
     }
 
     @OnScheduled
-    public void setup(final ProcessContext context) {
-        String configBody = context.getProperty(MIME_CONFIG_BODY).getValue();
-        String configFile = context.getProperty(MIME_CONFIG_FILE).evaluateAttributeExpressions().getValue();
+    public void setup(final ProcessContext context) throws IOException {
+        String configStrategy = context.getProperty(CONFIG_STRATEGY).getValue();
 
-        if (configBody == null && configFile == null){
+        if (configStrategy.equals(PRESET.getValue())){
             this.detector = config.getDetector();
             this.mimeTypes = config.getMimeRepository();
-        } else if (configBody != null) {
-            try {
-                this.detector = MimeTypesFactory.create(new ByteArrayInputStream(configBody.getBytes()));
-                this.mimeTypes = (MimeTypes)this.detector;
-            } catch (Exception e) {
-                context.yield();
-                throw new ProcessException("Failed to load config body", e);
-            }
-
         } else {
-            try (final FileInputStream fis = new FileInputStream(configFile);
-                 final InputStream bis = new BufferedInputStream(fis)) {
-                this.detector = MimeTypesFactory.create(bis);
-                this.mimeTypes = (MimeTypes)this.detector;
-            } catch (Exception e) {
-                context.yield();
-                throw new ProcessException("Failed to load config file", e);
-            }
+            setCustomMimeTypes(configStrategy, context);
         }
     }
 
+    private void setCustomMimeTypes(String configStrategy, ProcessContext context) throws IOException {
+        String configBody = context.getProperty(MIME_CONFIG_BODY).getValue();
+        String configFile = context.getProperty(MIME_CONFIG_FILE).evaluateAttributeExpressions().getValue();
+
+        try (final InputStream customInputStream = configBody != null ? new ByteArrayInputStream(configBody.getBytes()) : new FileInputStream(configFile);
+             final InputStream nifiInputStream = getClass().getClassLoader().getResourceAsStream("org/apache/tika/mime/custom-mimetypes.xml");
+             final InputStream tikaInputStream = MimeTypes.class.getClassLoader().getResourceAsStream("org/apache/tika/mime/tika-mimetypes.xml")) {
```

Comment:
These lines should be moved into the `else` block since they are not used for the `REPLACE` strategy.